mungegithub/mungers/submit-queue.go

```diff
@@ -647,6 +657,34 @@ func (sq *SubmitQueue) getQueueStatus() []byte {
 	return sq.marshal(status)
 }
 
+// getMetrics returns a json representation of important stats about the
+// submit queue's health, for external monitoring/graphing/alerting.
+func (sq *SubmitQueue) getMetrics() []byte {
+	sq.Lock()
+	defer sq.Unlock()
+
+	status := sq.e2e.GetBuildStatus()
+	blocked := false
+	for _, state := range status {
+		if state.Status != "Stable" {
+			blocked = true
+		}
+	}
```

Comment:
`sq.e2e.GCSWeakStable` does synchronous requests to GCS buckets, which is not what we want in this simple health check. Checking the build status like this replicates the return value of `GCSWeakStable` using the cached information.